pallets/sgx-account-linker/src/tests.rs

```diff
@@ -0,0 +1,231 @@
+		let block_number: u32 = 99999;
+		let layer_one_blocknumber: u32 = 10;
+
+		let mut gen = Random{};
+		let mut expected_vec = Vec::new();
+
+		for i in 0..(MAX_ETH_LINKS) {
+
+			let key_pair = gen.generate();
+
+			let msg = generate_msg(&account, block_number + i as u32);
+			let sig = generate_sig(&key_pair, &msg);
+
+			assert_ok!(SgxAccountLinker::do_link_eth(
+				account.clone(),
+				i as u32,
+				key_pair.address().to_fixed_bytes(),
+				block_number + i as u32,
+				layer_one_blocknumber,
+				sig
+			));
+
+			assert_eq!(SgxAccountLinker::eth_addresses(&account).len(), i+1);
+			expected_vec.push(key_pair.address().to_fixed_bytes());
+			assert_eq!(
+					events(),
+					[
+						Event::SgxAccountLinker(crate::Event::EthAddressLinked(account.clone(), key_pair.address().to_fixed_bytes().to_vec())),
+					]
+				);
+		}
+		assert_eq!(SgxAccountLinker::eth_addresses(&account), expected_vec);
+	});
+}
+
+#[test]
+fn test_update_eth_address() {
+	new_test_ext().execute_with(|| {
+
+		let account: AccountId32 = AccountId32::from([40u8; 32]);
+		let block_number: u32 = 99999;
+		let layer_one_blocknumber: u32 = 10;
+
+		let mut gen = Random{};
+		for i in 0..(MAX_ETH_LINKS) {
+			let key_pair = gen.generate();
+			let msg = generate_msg(&account, block_number + i as u32);
+			let sig = generate_sig(&key_pair, &msg);
+
+			assert_ok!(SgxAccountLinker::do_link_eth(
+				account.clone(),
+				i as u32,
+				key_pair.address().to_fixed_bytes(),
+				block_number + i as u32,
+				layer_one_blocknumber,
+				sig
+			));
+		}
+
+		let index: u32 = 2 as u32;
+		// Retrieve previous addr
+		let addr_before_update =  SgxAccountLinker::eth_addresses(&account)[index as usize];
+		// Update addr at slot `index`
+		let key_pair = gen.generate();
+		let block_number = block_number + 9 as u32;
+		let msg = generate_msg(&account, block_number);
+		let sig = generate_sig(&key_pair, &msg);
+
+		assert_ok!(SgxAccountLinker::do_link_eth(
+			account.clone(),
+			index,
+			key_pair.address().to_fixed_bytes(),
+			block_number,
+			layer_one_blocknumber,
+			sig
+		));
+
+		let updated_addr =  SgxAccountLinker::eth_addresses(&account)[index as usize];
+		assert_ne!(updated_addr, addr_before_update);
+		assert_eq!(updated_addr, key_pair.address().to_fixed_bytes());
+	});
+}
+
+
+#[test]
+fn test_eth_address_pool_overflow() {
+	new_test_ext().execute_with(|| {
+
+		let account: AccountId32 = AccountId32::from([113u8; 32]);
+		let block_number: u32 = 99999;
+		let layer_one_blocknumber: u32 = 10;
+
+		let mut gen = Random{};
+		let mut expected_vec = Vec::new();
+
+		for index in 0..(MAX_ETH_LINKS*2) {
+			let key_pair = gen.generate();
+
+			let msg = generate_msg(&account, block_number);
+			let sig = generate_sig(&key_pair, &msg);
+
+			assert_ok!(SgxAccountLinker::do_link_eth(
+				account.clone(),
+				index as u32,
+				key_pair.address().to_fixed_bytes(),
+				block_number,
+				layer_one_blocknumber,
+				sig
+			));
+
+			if index < MAX_ETH_LINKS {
```

Comment:
This piece of code is relatively similar to the original code of `do_link_eth`. So there's a danger of copying the error done in the code into the test. I think a seperate for loop to genereate the vector you're expecting would be more thinking-error safe. Something like:
```rust
for index in 0..(MAX_ETH_LINKS) {
    expected_vec.push(key_pair.address().to_fixed_bytes());
}
expected_vec[MAX_ETH_LINKS-1] = key_pair.address().to_fixed_bytes();

```